docs/competitive-programming/factorial-of-large-number.mdx

```diff
@@ -0,0 +1,201 @@
+---
+title: Factorial of very large number 
+slug: /competitive-programming/factorial-of-large-number
+---
+
+## Introduction
+
+Factorial of a non-negative number `n` is the product of all integers less than or equal to `n`.
+But when you have a very large `n` say 100, it's factorial has `158` digits, which is not possible to store even if 
+you use long long int. 
+
+
+## Explanation
+
+For a non-negative integer say `n`,the factorial of `n` is defined as follows:
+
+```
+Factorial(n) = n * (n - 1) * (n - 2) * . . . . . 3  * 2 * 1
+```
+
+Or
+
+```
+Factorial(n) = n * Factorial(n - 1)
+```
+
+Examples: 
+
+```
+Factorial(4) = 4 * 3 * 2 * 1 = 24 = 4 * Factorial(3)
+Factorial(7) = 7 * 6 * 5 * 4 * 3 * 2 * 1 = 5040 = 7 * Factorial(6)
+```
+
+## Algorithm
+
+```
+largeFactorial(n):
+	Set rsize to 1
+  	FOR x=2 to n
+		Set rsize to multiply(x, res, rsize)
+	ENDFOR
+	Print factorial of the number
+
+multiply(n,res,rsize):
+	Set carry to 0
+  	FOR i=0 to rsize-1
+		Set product to res[i] * x + carry
+		Set res[i] to product % 10
+		Set carry to product/10
+	ENDFOR
+	WHILE carry exists
+		push carry%10 to res
+		divide carry by 10
+		increment rsize
+	ENDWHILE
+	RETURN rsize
```

Comment:
We don't need pseudo-code here. An actual text-based explanation would be appreciated.

Reply:
Sure thing.